seatunnel-formats/seatunnel-format-text/src/main/java/org/apache/seatunnel/format/text/TextDeserializationSchema.java

```diff
@@ -55,13 +58,10 @@ public class TextDeserializationSchema implements DeserializationSchema<SeaTunne
     @Override
     public SeaTunnelRow deserialize(byte[] message) throws IOException {
         String content = new String(message);
-        String[] splits = content.split(delimiter);
-        if (seaTunnelRowType.getTotalFields() != splits.length) {
-            throw new IndexOutOfBoundsException("The data does not match the configured schema information, please check");
-        }
-        Object[] objects = new Object[splits.length];
-        for (int i = 0; i < splits.length; i++) {
-            objects[i] = convert(splits[i], seaTunnelRowType.getFieldType(i));
+        Map<Integer, String> splitsMap = splitLineBySeaTunnelRowType(content, seaTunnelRowType);
+        Object[] objects = new Object[splitsMap.size()];
```

Comment:
May need to check the data in this way the deserialization are legal. An exception needs to be thrown if it is illegal.

Reply:
> May need to check the data in this way the deserialization are legal. An exception needs to be thrown if it is illegal.

If a row type was existed in SeaTunnelRow, we also need count the total field of it either, so it doesn't make sense to just use the total number of splits and the total number of fields. Of course, if you think necessary to judge the total number of fileds and the total number splits, we can write a recursive method to count the total number of fields.